[WC-381] Widget generator:  fix NPM v7.x.x peer dependency install issue

IIsaku opened the pull request:
## Checklist
- Contains script tests ✅ 
- Contains breaking changes ❌
- Contains Atlas changes ❌
- Compatible with: MX 9️⃣

## This PR contains
- [x] Bug fix
- [ ] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Other (describe)

## What is the purpose of this PR?
This PR fixes failing `npm install` executions when scaffolding a widget project with the widget generator with NPM v7.x.x. It also explains users in the readme of generated widget projects how to install dependencies with NPM v7.x.x. 

These failures occur due to NPM v7.x.x installing peer dependencies inside the `node_modules` folder. Not all dependencies inside pluggable-widgets-tools share the same React dependency leading into an unresolvable dependency tree.

## Relevant changes
- Execute `npm install` with the command line flag `--legacy-peer-deps`. This ensures pluggable-widgets-tools dependencies install correctly with NPM v7.x.x. The flag is ignored by lower versions of NPM.
- Add in the readme template details about the development steps including installing dependencies with NPM v7.x.x.
- Set required node version for generator and PIW tools to v12 or higher.
- Note in the readme file for PIW tools that installation with NPM v7.x.x requires the flag.

## What should be covered while testing?
- Scaffold a widget project with the generator with npm v6 and v7. Verify that both succeed.

diego-antonelli commented on packages/tools/generator-widget/README.md:
````diff
@@ -15,7 +15,13 @@ This generator uses the Yeoman scaffolding tool to let you quickly create a [Men
+    ```bash
+     npm install -g npm@"<7"
+    ```
+
````
Lets add a warning for users already with npm 7 to use `--legacy-peer-deps` combined with npm install

diego-antonelli replied:
Same comment needs to be added in piw-tools README

IIsaku replied:
Changed it

Keraito reviewed (dismissed):
Assuming you address Diego's feedback, everything looks good 👍

IIsaku committed:
fix: support NPM v7.x.x with widget tooling

- Execute npm install with the command line flag
"--legacy-peer-deps". This ensures pluggable-widgets-tools
dependencies install correctly with NPM v7.x.x
and is ignored by lower versions.
- Add in the readme template details about the development
including installing dependencies with NPM v7.x.x
- Set required node version for generator and PIW tools
to v12 or higher.
- Note in the readme file for PIW tools that NPM v7.x.x
requires the flag.

IIsaku committed:
chore: add changelogs

Keraito commented on packages/tools/generator-widget/CHANGELOG.md:
```diff
@@ -0,0 +1,14 @@
+
+### Fixed
+- Fixes failing `npm install` executions when scaffolding a widget project with the widget generator with NPM v7.x.x.
+- Explains users in the readme of generated widget project how to install dependencies with NPM v7.x.x.
```
```suggestion
- Explains users in the README of generated widget project how to install dependencies with NPM v7.x.x.
```
I think this is the usual convention

IIsaku replied:
Fixed

Keraito commented on packages/tools/generator-widget/generators/app/templates/commons/README.md.ejs:
```diff
@@ -14,4 +14,11 @@
-[specify contribute]
+
+1. Install NPM package dependencies by using: `npm install`. If you use NPM v7.x.x, which can be checked by executing `npm -v`, execute: `npm install --legacy-peer-deps`.
+1. Run `npm start` to watch for code changes. On every change:
+    - the widget will be bundled;
+    - the bundle will be included in a `dist` folder in the root directory of the project;
+    - the bundle will be included in the `deployment` and `widgets` folder of the Mendix test project.
```
I'm a bit confused. Is this the README template that will be added with the generator when creating new widgets? I can understand adding the first line about NPM >= v7, but why are we suddenly adding this `npm start` part?

IIsaku replied:
To give the section a bit more sense. Just having that note about NPM 7 feels like an incomplete section.

Keraito commented on packages/tools/generator-widget/CHANGELOG.md:
```diff
@@ -0,0 +1,14 @@
+
+## [Unreleased]
+## Changed
+- Set required node version to v12 or higher.
```
Internal question: is this a required for NPM >= v7?

IIsaku replied:
Node 12 was always required by PIW tools package. Moreover, Node 10 is EOL at the end of this month.

Keraito replied:
So this is more like a fly-by in this PR to update the node dependency rather than being part of the npm v7 fixes?

IIsaku replied:
Correct

Keraito commented on packages/tools/generator-widget/generators/app/index.js:
```diff
@@ -61,7 +61,7 @@ class MxGenerator extends Generator {
     install() {
         this.log(text.INSTALL_FINISH_MSG);
-        this.npmInstall();
+        this.npmInstall(undefined, { legacyPeerDeps: true });
```
I'm not entirely sure if this is a CLI script, but maybe it's an improvement to make this a command flag that defaults to `true` so users can also opt out? Or is that not relevant for this?

IIsaku replied:
It's not relevant. NPM versions below v7 will ignore this flag. For v7 or higher it's always required. When PIW tools dependencies get aligned we can remove this argument again.

Keraito replied:
okay, clear 👍

Keraito commented on packages/tools/pluggable-widgets-tools/CHANGELOG.md:
```diff
@@ -0,0 +1,13 @@
+- Set required node version to v12 or higher.
+
+### Fixed
+- Note in the readme file that installation with NPM v7.x.x requires the flag `--legacy-peer-deps`.
```
```suggestion
- Note in the README file that installation with NPM v7.x.x requires the flag `--legacy-peer-deps`.
```

IIsaku replied:
Fixed

Keraito reviewed (commented):
Few remarks

IIsaku committed:
chore(generator-widget): improve CHANGELOG.md

IIsaku committed:
chore: use uppercase README
